tests/OptimizelyTest.php

```diff
@@ -1730,4 +1731,498 @@ public function testGetVariationBucketingIdAttribute()
         $this->assertEquals(null, $variationKey, sprintf('Invalid variation key "%s" for getVariation with bucketing ID "%s".', $variationKey, $this->testBucketingIdControl));
     }
 
+    public function testIsFeatureEnabledGivenInvalidDataFile(){
+        $optlyObject = new Optimizely('Random datafile', null, $this->loggerMock);
+
+        $this->expectOutputRegex("/Datafile has invalid format. Failing 'isFeatureEnabled'./");
+        $optlyObject->isFeatureEnabled("boolean_feature", "user_id");
+    }
+
+    public function testIsFeatureEnabledGivenInvalidArguments(){
+        // should return null and log a message when feature flag key is empty
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::ERROR,  "Feature Flag key cannot be empty.");
+
+        $this->assertSame($this->optimizelyObject->isFeatureEnabled("","user_id"), null);
+
+        // should return null and log a message when feature flag key is null
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::ERROR,  "Feature Flag key cannot be empty.");
+
+        $this->assertSame($this->optimizelyObject->isFeatureEnabled(null,"user_id"), null);
+
+        // should return null and log a message when user id is empty
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::ERROR,  "User ID cannot be empty.");
+
+        $this->assertSame($this->optimizelyObject->isFeatureEnabled("boolean_feature", ""), null);
+
+        // should return null and log a message when user id is null
+        $this->loggerMock->expects($this->at(0))
```

Comment:
should it be at 0 or 1?

Reply:
Once loggermock has been called. It again starts from 0